server/datasources/cachedMongooseDataSource.ts

```diff
@@ -13,74 +12,33 @@ export default class CachedMongooseDataSource<DocumentType extends Document> ext
   }
 
   getById(id: Types.ObjectId): DocumentType {
-    return this.cache.getData().filter(request => request._id && request._id.equals(id))[0]
+    const res = this.cache.getData().filter(request => request._id && request._id.equals(id))
+    if(res.length === 0) {
+      throw new Error(`Mongoose ${this.cache.name} ObjectId not found`)
+    }
+    return res[0]
   }
 
   getAll(): Array<DocumentType> {
     return this.cache.getData()
   }
 
-  async create(inputObject: Document): Promise<ServerResponseInterface> {
+  async create(inputObject: Document): Promise<Document> {
     const newObject = new this.cache.model(inputObject)
     const promise = await newObject.save()
-      .then(res => {
-        return {
-          "success": true,
-          "message": `${this.cache.name} successfully created`,
-          "id": res._id
-        }
-      })
-      .catch(error => {
-        console.log(error)
-        return {
-          "success": false,
-          "message": error._message,
-          "id": null 
-        }
-      })
     return promise
   }
 
-  async update(inputObject: Document): Promise<ServerResponseInterface> {
+  async update(inputObject: Document, session: ClientSession): Promise<Document> {
     if(!inputObject.id) {
       throw new UserInputError('Missing argument value', { argumentName: 'id' })
     }
-    const promise = await this.cache.model.findByIdAndUpdate(inputObject.id.toString(), inputObject)
-      .then(res => {
-        return {
-          "success": true,
-          "message": `${this.cache.name} successfully updated`,
-          "id": inputObject.id
-        }
-      })
-      .catch(error => {
-        console.log(error)
-        return {
-          "success": false,
```

Comment:
Making sure this is intended. It appears that you removed the error message returned everywhere but didnt put it in the resolver. Atm this ServerResponseInterface doesn't appear to be anywhere.

Reply:
i removed it cuz I thought the error response given by graphql is more informative than the one we had. i could add it back if we feel otherwise.

Reply:
We should standardize our output. Maybe it should be decided if we should remove the success message and instead return the created/updated object always? Not sure what that may look like